repos/system_upgrade/el7toel8/actors/unsupportedupgradecheck/actor.py

```diff
@@ -0,0 +1,62 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class UnsupportedUpgradeCheck(Actor):
+    """
+    Checks enviroment variables and produces a warning report if the upgrade is unsupported.
+
+    Upgrade is unsupported if any LEAPP_DEVEL_* variable is used or an experimental actor is enabled.
+    This can be overridden by setting the variable LEAPP_UNSUPPORTED (at user's own risk).
+    """
+
+    name = 'unsupported_upgrade_check'
+    consumes = ()
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        conf = self.configuration
+        leapp_vars = conf.leapp_env_vars
+        devel_vars = [v for v in leapp_vars if v.name.startswith('LEAPP_DEVEL_')]
+        experimental = bool([v for v in leapp_vars if v.name == 'LEAPP_EXPERIMENTAL' and v.value == '1'])
```

Comment:
LEAPP_EXPERIMENTAL env var is not being set anywhere. So if you use e.g. `--whitelist-experimental removeleftoverpackages` the upgrade is not inhibited.

Reply:
Ah, I overlooked the https://github.com/oamg/leapp/pull/577. Nevermind then.